### ISSUE TITLE
[v2.2.1] Release Tracker 

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#119295]
### 🐛 Describe the bug

This issue is for tracking cherry-picks to the release branch. Following is [release branch](https://github.com/pytorch/pytorch/tree/release/2.2) for the 2.2.1 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 2/14): Cherry-pick post deadline (End of day 5PM PST)
* Phase 2 (after 2/14):  Perform extended integration/stability/performance testing based on Release Candidate builds.

**Only issues that have ‘cherry-picks’ in this tracker will be considered for the release.**

## Cherry-Pick Criteria

**Phase 1 (until 2/14):**

The Releng team relies on the cherry pick process to manage risk to release quality, i.e. by porting a small set of commit from trunk that are "must-have" into the release branch, we limit the change to the minimal to address pressing issues. Thus, not everything a developer land into the trunk will make it into the release. So, please consider the criteria below and follow the cherry picking process. Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent release (e.g. 2.2.0 for 2.2.1 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Low risk critical fixes for: silent correctness, backwards compatibility, crashes, deadlocks, (large) memory leaks
3. Fixes to new features being introduced in 2.2.0 release
4. Documentation improvements
5. Release branch specific changes (e.g. blocking ci fixes, change version identifiers)

Any other change requires special dispensation from the release managers (currently @atalman, @huydhn, @osalpekar, @malfet). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain. 

**Phase 2 (after 2/14):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/2.1).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/2.2
    git checkout release/2.2
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/2.2' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/2.2
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed trunk PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

See [HUD 2.2](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F2.2/1?per_page=50) 

### Versions

2.2.1

**Comment by huydhn** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/builder/pull/1697

Link to release branch PR:
* https://github.com/pytorch/builder/pull/1699

Criteria Category:
* CI fixes https://github.com/pytorch/pytorch/issues/114721 to enable OpenMP backend for MacOS 3.11 wheel.

**Comment by antoniojkim** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/117653 

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119301

Criteria Category:
* Low risk critical fix for unrecoverable crash via abort

**Comment by snadampal** [comment]
Link to master PR (if applicable):
- https://github.com/pytorch/builder/pull/1696

Link to 2.2 release branch PR:
- https://github.com/pytorch/builder/pull/1700

Criteria Category:
Performance regression.
This PR fixes the regression introduced when the aarch64 linux wheel builder was moved from local openblas builds to conda package during PyTorch 2.1.

**Comment by tringwald** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/118465
* https://github.com/pytorch/pytorch/pull/118315
* https://github.com/pytorch/pytorch/pull/115041

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119313
* https://github.com/pytorch/pytorch/pull/119315
* https://github.com/pytorch/pytorch/pull/119316

Criteria Category:
* Doc fixes

**Comment by mvpatel2000** [comment]
Dtensor / device mesh is currently broken on 2.2 and has serious silent correctness errors. These were fixed by teh following PRs:
- https://github.com/pytorch/pytorch/pull/116559
- https://github.com/pytorch/pytorch/pull/117020
- https://github.com/pytorch/pytorch/pull/116792

Related to distributed checkpointing:
- https://github.com/pytorch/pytorch/pull/115592

Related dtensor sync_module_states bug:
- https://github.com/pytorch/pytorch/pull/117336

These are all low-risk PRs and fix various issues, eg silent correctness. 

CC: @chauhang

**Comment by mdevino** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/119214

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119388

Criteria Category:
Documentation Improvements - this PR fixes a typo in the `torch.frombuffer()` function documentation.

**Comment by mdevino** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/119428

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119505

Criteria Category:
Documentation Improvements - this PR fixes a typo in the contribution guide

**Comment by huydhn** [comment]
Link to landed master PR (if applicable):
* https://github.com/pytorch/builder/pull/1698

Link to release branch PR:
* https://github.com/pytorch/builder/pull/1701

Criteria Category:
* CI fixes to build conda MacOS due to https://github.com/conda/conda-build/issues/5167.  We might not need this as the issue has been fixed in conda 24.1.1 patch release.  This is standing by in case RC build is failing.

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/117416

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119560

Criteria Category:
* CI fixes use oidc when uploading binaries

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/117648
* https://github.com/pytorch/pytorch/pull/117706

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119564

Criteria Category:
* CI fixes use oidc when uploading triton wheels

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/117160
* https://github.com/pytorch/pytorch/pull/117422

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119565

Criteria Category:
* CI fixes use oidc for rocm

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/117290
* https://github.com/pytorch/pytorch/pull/117371

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119568

Criteria Category:
* CI fixes use oidc

**Comment by mvpatel2000** [comment]
Fixes sharded checkpoint bug which causes OOM on large GPU counts:

https://github.com/pytorch/pytorch/pull/117799

CC: @chauhang @fegin

**Comment by Skylion007** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/116426
Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119687
Criteria Category:
* Silent Correctness

**Comment by Skylion007** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/116792

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119688

Criteria Category:
* Silent Correctness

**Comment by Skylion007** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/115592

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119691

Criteria Category:
* Checkpoint loading bug in DCP: Fix to new feature / regression fix.

**Comment by huydhn** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/test-infra/pull/4870
* https://github.com/pytorch/test-infra/pull/4877
* https://github.com/pytorch/test-infra/pull/4882
* https://github.com/pytorch/test-infra/pull/4886
* https://github.com/pytorch/test-infra/pull/4891
* https://github.com/pytorch/test-infra/pull/4893
* https://github.com/pytorch/test-infra/pull/4894
* https://github.com/pytorch/test-infra/pull/4901

Link to release branch PR:
* https://github.com/pytorch/test-infra/pull/4945

Criteria Category:
* CI cherry-picks to use OIDC in the release workflows

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/test-infra/pull/4946

Criteria Category:
* CI release only, bump versions

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/118860

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119769

Criteria Category:
* Critical FIx  Windows

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/119743

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119770

Criteria Category:
* PyPi Metadata fix

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/118057

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119773

Criteria Category:
* Critical Regression

**Comment by fegin** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/119573

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119807

Criteria Category:
* Low risk critical fixes

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/builder/pull/1703

Link to release branch PR:
* https://github.com/pytorch/builder/pull/1704

Criteria Category:
* Critical CI fix - Triton

**Comment by mvpatel2000** [comment]
Link to landed trunk PR (if applicable):
- https://github.com/pytorch/pytorch/pull/112435
- https://github.com/pytorch/pytorch/pull/118620
- https://github.com/pytorch/pytorch/pull/119064
- https://github.com/pytorch/pytorch/pull/118638
- https://github.com/pytorch/pytorch/pull/119481

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/119819

Criteria Category:
- Fixes to new features being introduced in 2.2.0 release

**Comment by wanchaol** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/116151

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119821

Criteria Category:
* Documentation Improvements

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/118591

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119894

Criteria Category:
* Critical FIx on V100, regression against 2.1.0 release

**Comment by atalman** [comment]
Link to landed master PR (if applicable):
[Properly preserve SymInt input invariant when splitting graphs #117406](https://github.com/pytorch/pytorch/pull/117406)

Link to release branch PR:
[Properly preserve SymInt input invariant when splitting graphs (#117406)](https://github.com/pytorch/pytorch/pull/118067)

Criteria Category:
Crashes - Functional failure in case of DDPOptimizer and dynamic shape

**Comment by mvpatel2000** [comment]
Link to landed master PR:
https://github.com/pytorch/pytorch/pull/117453

Link to release branch PR:
https://github.com/pytorch/pytorch/pull/119916

Criteria Category:
Fixes to new features being introduced in 2.2.0 release -- minor fix for argument not propagated in new API introduced in 2.2

**Comment by wanchaol** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/119659

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119941

Criteria Category:
* regressions against the most recent release, silent correctness

**Comment by Vins33** [comment]
Hi guys, could you tell me if the next release will support cuda12.4?

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/119750

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/119988

Criteria Category:
* Critical regression

**Comment by atalman** [comment]
Please Note We don't accept cherry-picks anymore.  We are currently in Phase 2 (after 2/14).

**Comment by huydhn** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/test-infra/pull/4958

Link to release branch PR:
* https://github.com/pytorch/test-infra/pull/4962

Criteria Category:
* CI fix to ignore 3.12 data binary which doesn't exist atm

---
@huydhn merged

**Comment by atalman** [comment]
Closing this. Completed
